examples/seismic/model.py

```diff
@@ -397,3 +397,105 @@ def critical_dt(self):
         # for further details:
         dt = .85*np.min(self.spacing) / (np.sqrt(self.grid.dim)*self.maxvp)
         return self.dtype("%.3e" % dt)
+
+class ModelViscoacoustic(GenericModel):
+    """
+    The physical model used in seismic inversion processes.
+
+    Parameters
+    ----------
+    origin : tuple of floats
+        Origin of the model in m as a tuple in (x,y,z) order.
+    spacing : tuple of floats
+        Grid size in m as a Tuple in (x,y,z) order.
+    shape : tuple of int
+        Number of grid points size in (x,y,z) order.
+    space_order : int
+        Order of the spatial stencil discretisation.
+    vp : array_like or float
+        Velocity in km/s.
+    rho : float or array, optional
+        Density in kg/cm^3 (rho=1 for water).
+    qp : float or array, optional
+        P-wave quality factor (dimensionless).
+    f0: float or Scalar, optional
+        The dominant frequency
+    nbl : int, optional
+        The number of absorbin layers for boundary damping.
+    dtype : np.float32 or np.float64
+        Defaults to 32.
+
+    The `ModelViscoacoustic` provides two symbolic data objects for the
+    creation of seismic wave propagation operators:
+
+    m : array_like or float
+        The square slowness of the wave.
+    damp : Function
+        The damping field for absorbing boundary condition.
+    """
+    def __init__(self, origin, spacing, shape, space_order, vp, qp, rho, nbl=20, 
+                 subdomains=(), dtype=np.float32):
+        super(ModelViscoacoustic, self).__init__(origin, spacing, shape, space_order,
+                                                nbl=nbl, subdomains=subdomains, 
+                                                dtype=dtype, damp_mask=True)
+
+        self._vp = self._gen_phys_param(vp, 'vp', space_order)
+
+        self.qp = self._gen_phys_param(qp, 'qp', space_order, is_param=True)
+
+        self.rho = self._gen_phys_param(rho, 'rho', space_order, is_param=True)
+
+        self.irho = self._gen_phys_param(1./rho, 'irho', space_order, is_param=True)
+        
+    @property
+    def _max_vp(self):
+        return mmax(self.vp)
+
+    @property
+    def critical_dt(self):
+        """
+        Critical computational time step value from the CFL condition.
+        """
+        # For a fixed time order this number decreases as the space order increases.
+        #
+        # The CFL condtion is then given by
+        # dt <= coeff * h / (max(velocity))
+        coeff = 0.38 if len(self.shape) == 3 else 0.42
+        dt = self.dtype(coeff * np.min(self.spacing) / (self._max_vp))
+        return self.dtype("%.3e" % dt)
+
+    @property
```

Comment:
All these are in `Model` so would inherit from it rather thant repeat code.

Reply:
Okay. It really isn't necessary. We will fix it.

Reply:
At this moment the code is repeated because the critical dt is the same as acoustic class.
We understood that we should inherit the viscoacoustic class from the generic model, instead of the model. This choice is based on the inheritance convention for other classes (e.g elastic). 

We don't found any way to replace the generic model to model, because the damp_mask variable is not available in the standard model, at least without any modification.

Reply:
>damp_mask variable is not available in the standard model,

Yes it is

Reply:
@mloubout, I know that the model is inherited from the generic model (which has the damp_mask attribute). But in the model builder, damp_mask is not passed to the super (generic model), therefore, damp_mask is always false. Could I add the damp_mask attribute to the __init__ model, with the default False, and send that attribute to the super (generic model)?  Or is there another solution design? Thanks!

Reply:
Mathias, I know that model is inherited from the generic model (that has damp_mask attribute). But in the model constructor, the damp_mask is not passed to the super (generic model), so always damp_mask is false. Could I add the damp_mask attribute to the model __init__, with default False, and send this attribute to the super (generic model)?  Or is there another solution design? Thanks!

Reply:
>Could I add the damp_mask attribute to the model init,  with default False

Yes, this way can inherit from it and only have the extra qp, rho in the visco model (rho should go in Model as only qp is visco but that's for another time)

Reply:
Perfect, we will push this modification. Thank you.

Reply:
Done.